HW_6_Files/bio_files_processor.py

```diff
@@ -0,0 +1,172 @@
+import os
+
+OUTPUT_FASTA_SUFFIX = '_output_file'
+FASTA_FILE_EXT = '.fasta'
+
+LOCUS_GBK_KEYWORD = 'LOCUS'
+FEATURES_GBK_KEYWORD = 'FEATURES'
+ORIGIN_GBK_KEYWORD = 'ORIGIN'
+CDS_GBK_KEYWORD = 'CDS'
+END_OF_PART_GBK_KEYWORD = '//'
+
+GENE_CDS_KEYWORD = '/gene'
+TRANSLATION_CDS_KEYWORD = '/translation'
+
+def get_last_dict_value(input_dict: dict):
+    keys = list(input_dict.keys())
+    return input_dict[keys[-1]]
+
+
+def convert_multiline_fasta_to_oneline(input_fasta: str,
+                                       output_fasta: str = ''):
+    """
+       Converts a multi-line FASTA file to a one-line FASTA format.
+
+       Args:
+           input_fasta (str): The path to the input multi-line FASTA file.
+           output_fasta (str, optional): The path to the output one-line FASTA
+           file. If not provided, it will be generated based on the input file.
+
+       Raises:
+           ValueError: If the input FASTA file does not exist.
+
+       Note:
+           The function reads the input multi-line FASTA file and converts it
+           to a one-line FASTA file, where each sequence is on a single line.
+
+       """
+    if not os.path.exists(input_fasta):
+        raise ValueError(f'Your {input_fasta} doesn\'t exist')
+    if output_fasta != '':
+        if not output_fasta.endswith(FASTA_FILE_EXT):
+            output_fasta += FASTA_FILE_EXT
+    else:
+        output_fasta = os.path.join(os.path.dirname(input_fasta),
+                                    os.path.splitext
+                                    (input_fasta)[0] +
+                                    OUTPUT_FASTA_SUFFIX + FASTA_FILE_EXT)
+    with open(input_fasta) as input_fasta_file:
+        with open(output_fasta, mode="w") as output_fasta_file:
```

Comment:
можно избежать вложенности конструкций, открыв два фаза за один раз)
```suggestion
    with open(input_fasta, "r") as input_fasta_file, open(output_fasta, mode="w") as output_fasta_file:
```